### ISSUE TITLE
OADP-732: Enable ability to use private registry for custom plugins (add image pull secrets)

### ISSUE THREAD

**Issue by PendaGTP** [openshift/oadp-operator#739]
**Is your feature request related to a problem? Please describe.**
I want to use a custom plugins, the image of it is located on a private container registry.
It is not possible to add an imagePullSecret to the Velero deployment.

**Describe the solution you'd like**
Allow to set the imagePullSecrets property.

**Describe alternatives you've considered**
If the configuration of the imagePullSecrets property is not possible, it could be nice to indicate to the controller to ignore the modifications of this property, this one could be modified by a mutation at admission.
This is probably not the best idea, I'm not very inspired.

**Comment by PendaGTP** [comment]
@kaovilai, yes I can create a secret that contains the information needed to authenticate to the private registry but how do I tell to OADP Operator to use this secret (and thus add the secret to imagePullSecret of the Velero deployment)?
If I modify the deployment of Velero, the operator undo my modifications, which I understand.

Without this possibility, I don't see how to use a custom plugin which is located on a private registry (https://github.com/openshift/oadp-operator/blob/master/docs/config/plugins.md).

**Comment by kaovilai** [comment]
Ok thanks. Reading it again this is a feature we have to implement.

**Comment by kaovilai** [comment]
Tracked in https://issues.redhat.com/browse/OADP-732
